src/app/js/formats/vega-lite/component/vega-component/VegaLiteComponent.js

```diff
@@ -0,0 +1,49 @@
+import { Vega } from 'react-vega';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { isAdmin } from '../../../../user';
+
+function CustomActionVegaLite(props) {
+    let actions;
+    if (isAdmin(props.user)) {
+        actions = {
+            export: {
+                svg: true,
+                png: true,
+            },
+            source: true,
+            compiled: true,
+            editor: true,
+        };
+    } else {
+        actions = {
+            export: {
+                svg: true,
+                png: true,
+            },
+            source: false,
+            compiled: false,
+            editor: false,
+        };
+    }
+
+    let spec = props.spec;
```

Comment:
use const instead of let, and I think this syntax is more conventional :

const { spec } = props;

see https://evertpot.com/javascript-let-const/ & https://exploringjs.com/es6/ch_destructuring.html